compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -465,6 +468,54 @@ fn expand_preparsed_asm(ecx: &mut ExtCtxt<'_>, sp: Span, args: AsmArgs) -> P<ast
         for piece in unverified_pieces {
             match piece {
                 parse::Piece::String(s) => {
+                    if let Some(idx) = s.find(".intel_syntax") {
+                        let mut end = idx + ".intel_syntax".len();
+                        if let Some(prefix_idx) = s.split_at(end).1.find("noprefix") {
+                            // Should be a space and it should be immediately after
+                            if prefix_idx == 1 {
+                                end += " noprefix".len();
+                            }
+                        }
+
+                        let syntax_span =
+                            template_span.from_inner(InnerSpan::new(idx + 1, end + 1));
+                        let mut err = ecx.struct_span_err(syntax_span, "intel sytnax is the default syntax, and trying to use this directive may cause issues");
```

Comment:
AFAICT would get printed on architectures other than x86, but `.intel_syntax` is only the default on x86.

You can get to the target through `ecx.sess.target` and `AsmArch` through `ecx.sess.asm_arch`. The default syntax is computed here:

https://github.com/rust-lang/rust/blob/cb2effd44e667d133e31ef334e30d10195218ce6/compiler/rustc_codegen_llvm/src/asm.rs#L280-L287

Reply:
How should attempts to use intel syntax on att syntax default systems be handled?  There's no equivalent syntax option.  Should it emit an error that just says that use is unsupported, and if there ever is an intel syntax option (which I think there should be) the error can be updated?

Reply:
An error will occur somewhere down the line. Macro expansion code does not need to concern itself with this problem.

With that in mind I think this macro expansion code should entirely ignore presence of `.intel_syntax` on systems that don't support this directive (i.e. anything that's not x86(_64)).